File: Aeds II/Est & Algo/C-like/Pesquisa em memória primária/PesquisaSequencial.c
#include <stdio.h>
#include <stdbool.h>

bool PsqSequencial(int array[], int len, int x) {
    bool resp = false;

    for (int i = 0; i < len; i++) {
        if (array[i] == x) {
            resp = true;
            break;
        }
    }

    return resp;
}

int main() {
    printf("Digite o número que deseja buscar no array: ");
    int x;
    scanf("%d", &x);

    int vet[] = {2, 3, 5, 7, 9, 11, 15, 17, 20, 21, 30, 43, 49, 70, 71, 82, 23, 34, 46, 55, 64, 72, 76, 87, 92, 98, 37, 53, 67, 79, 91, 94};
    int len = sizeof(vet) / sizeof(vet[0]);

    bool resp = PsqSequencial(vet, len, x);

    if (resp) {
        printf("Está presente no vetor\n");
    } else {
        printf("Não está presente no vetor\n");
    }

    return 0;
}

//Melhor caso: f(n) = 1 --> primeira posição
//Pior caso:   f(n) = n --> ultima posição

File: Matérias/Aeds II/Est & Algo/Java/Pesquisa em memória primaria/PesquisaSequencial.java
import java.util.*;

public class PesquisaSequencial
{
    public static boolean PsqSequencial(int[] array, int x)
    {
        boolean resp = false;
        int len = array.length; // array.length != array.length()

        for(int i = 0;  i < len; i++)
        {
            if(array[i] == x)
            {
                resp = true;
                i = len;
            }
        }

        return resp;
    }

    public static void main(String[] args)
    {
        Scanner scanf = new Scanner(System.in);

        System.out.println("Digite o número que deseja buscar no array: ");
        int x = scanf.nextInt();

        int[] vet = {2, 3, 5, 7, 9, 11, 15, 17, 20, 21, 30, 43, 49, 70, 71, 82, 23, 34, 46, 55, 64, 72, 76, 87, 92, 98, 37, 53, 67, 79, 91, 94};

        boolean resp = PsqSequencial(vet, x);

        if(resp)
        {
            System.out.println("Está presente no vetor");
        }
        else
        {
            System.out.println("Não está presente no vetor");
        }
        
        scanf.close();
    }
}

//Melhor caso: f(n) = 1 --> primeira posição
//Pior caso:   f(n) = n --> ultima posição